pkg/resource/compare/utils.go

```diff
@@ -0,0 +1,41 @@
+package compare
+
+import (
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"reflect"
+)
+
+type mapBuilder struct {
+	resourceMap map[reflect.Type][]resource.KubernetesResource
+}
+
+func NewMapBuilder() *mapBuilder {
+	this := &mapBuilder{resourceMap: make(map[reflect.Type][]resource.KubernetesResource)}
+	return this
+}
+
+func (this *mapBuilder) Map() map[reflect.Type][]resource.KubernetesResource {
+	return this.resourceMap
+}
+
+func (this *mapBuilder) SameTypeItems(items ...resource.KubernetesResource) *mapBuilder {
```

Comment:
I would just use one function `Items` that would reuse `Item`. I'd chose simplicity and safety over performance in this case.
Regarding naming. `Item` suggests it will _get_ an _item_, however this is not what the method does. That's why I suggest using `AddXXX`. Besides, to be more precise, it is adding a _resource_ so why not call it `AddResource` ? or even `Add`, because it's the only thing the API allows to add.

My final suggestion is that a single method `Add` that accepts varargs should be enough?
```
func (this *mapbuilder) Add(items ...resource.KubernetesResource) *mapBuilder {
  for index := range items {
    resourceType := reflect.ValueOf(item).Elem().Type()
￼    this.resourceMap[resourceType] = append(this.resourceMap[resourceType], item)
  }
  return this
}
```